connectors/es/client.py

```diff
@@ -74,6 +83,42 @@ def stop_waiting(self):
         self._keep_waiting = False
         self._sleeps.cancel()
 
+    async def has_license_enabled(self, license_):
```

Comment:
It feels strange that you return `boolean` and the license type from a method `has_license_enabled`. I understand you want to log the current license type in the message. WDYT if we make this method `license_type`:
```python
class License(Enum):
    ENTERPRISE = "enterprise"
    PLATINUM = "platinum"
    BASIC = "basic"
    TRIAL = "trial"
    EXPIRED = "expired"
    UNSET = None


async def license_type(self):
    """This method checks, whether a certain license or a more powerful license is enabled."""

    license_info = await self.client.license.get().get("license", {})
    license_type = license_info.get("type").lower()
    if license_info.get("status") == "expired":
        license_type = "expired"

    return License(license_type)



license_type = await self.connector_index.license_type()

if license_type not in (License.PLATINUM, License.ENTERPRISE):
    if license_type == License.EXPIRED:
        logger.error(
            f"Elasticsearch has an expired license. Skipping access control sync execution..."
        )
    else:
        logger.error(
            f"Elasticsearch doesn't use the license 'platinum', instead it uses '{license_type}'. Skipping access control sync execution..."
        )
    return
```

Reply:
Personally, I would like to keep the license check inside the method itself. This would avoid bugs, where someone calls the method at different places and forgets to check for more powerful licenses. If we hide this logic inside the method we guarantee consistent behavior.

WDYT?